package.json

```diff
@@ -18,7 +18,41 @@
         "url": "git+https://github.com/nick-w-nick/eslint-config-lintification.git"
     },
     "homepage": "https://github.com/nick-w-nick/eslint-config-lintification#readme",
-    "keywords": ["eslint", "lintification", "lint", "typescript", "nodejs", "backend"],
+    "keywords": [
+        "eslint",
+        "lintification",
+        "lint",
+        "typescript",
+        "nodejs",
+        "backend"
+    ],
     "author": "https://github.com/nick-w-nick",
-    "license": "MIT"
+    "license": "MIT",
+    "dependencies": {
+        "eslint": ">= 8.56.0",
+        "@stylistic/eslint-plugin": "2.10.1",
+        "@typescript-eslint/eslint-plugin": "6.7.4",
+        "@typescript-eslint/parser": "6.7.4",
+        "eslint-config-airbnb": "19.0.4",
+        "eslint-config-airbnb-base": "15.0.0",
+        "eslint-config-airbnb-typescript": "17.1.0",
+        "eslint-import-resolver-typescript": "3.6.1",
+        "eslint-plugin-file-progress": "1.3.0",
+        "eslint-plugin-import": "2.28.1",
+        "eslint-plugin-import-newlines": "1.3.4",
+        "eslint-plugin-jest": "27.4.2",
+        "eslint-plugin-newline-destructuring": "1.2.2",
+        "eslint-plugin-simple-import-sort": "10.0.0",
+        "typescript-eslint": "8.7.0"
+    },
+    "peerDependencies": {
+        "@stylistic/eslint-plugin": "2.10.1",
+        "eslint": ">= 8.56.0",
+        "eslint-plugin-file-progress": "1.3.0",
+        "eslint-plugin-import": "2.28.1",
+        "eslint-plugin-import-newlines": "1.3.4",
+        "eslint-plugin-jest": "27.4.2",
+        "eslint-plugin-newline-destructuring": "1.2.2",
+        "eslint-plugin-simple-import-sort": "10.0.0"
+    }
```

Comment:
Can you summarize why some have to be peer dependencies and others not?

Reply:
That's summarized in this thread: https://github.com/nick-w-nick/eslint-config-lintification/pull/1#discussion_r1845742004

Reply:
This part is most important: `plugins need to be peer dependencies`. When doing `npm link` locally, if something is a plugin, I needed it as a peer dependency and to be installed as a dev dependency in the consuming project. If it's not a plugin, I found it could be a regular dependency.

The one exception I found to that rule of thumb was `eslint-import-resolver-typescript`, which threw these errors in every file if it wasn't listed as a peer dependency:
```
1:1   error  Resolve error: typescript with invalid interface loaded as resolver      import/no-relative-packages
  1:27  error  Unexpected use of file extension "js" for "./path-to-your-file.js"  import/extensions
```

Though, what's strange is [eslint's own documentation](https://eslint.org/docs/latest/extend/shareable-configs#publishing-a-shareable-config) says to list it as a `dependency`, so it's unclear to me if this is just strange behavior coming from how `npm link` intersects with linting config packages or not.

Reply:
The documentation says to list `eslint` as a peer dependency (red), and make the plugins regular dependencies (green).

![Screenshot 2024-11-18 at 12 07 09 PM](https://github.com/user-attachments/assets/7805dd9d-3a67-4e62-9923-22f7a1b64660)

I can't totally tell if it means they need to be dependencies in the consuming project's package.json or if it means in ours, but it might work with just eslint in peer deps and the rest in regular deps.